src/main.go

```diff
@@ -47,6 +46,26 @@ type ErrorResponse struct {
 // Groups together HTTP handler functions and holds a reference to the application's config
 type handlers struct {
 	config *Config
+	client *http.Client
+}
+
+type authTransport struct {
+	transport http.RoundTripper
+	token     azcore.AccessToken
+}
+
+func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.token.ExpiresOn.Before(time.Now()) {
+		var err error
+		t.token, err = cred.GetToken(req.Context(), policy.TokenRequestOptions{
+			Scopes: []string{"https://cognitiveservices.azure.com/.default"},
+		})
+		if err != nil {
+			return nil, err
```

Comment:
The Roundtrip interface says that the body must be closed on error.
```suggestion
			req.Body.Close()
			return nil, err
```